pkg/wasm/types.go

```diff
@@ -116,12 +115,12 @@ func (s *ActionSet) EqualTo(other ActionSet) bool {
 }
 
 type RouteRuleConditions struct {
-	Hostnames []string    `json:"hostnames"`
-	Matches   []Predicate `json:"matches,omitempty"`
+	Hostnames  []string `json:"hostnames"`
+	Predicates []string `json:"predicates,omitempty"`
```

Comment:
Check out https://github.com/Kuadrant/kuadrant-operator/pull/976/files#diff-509b896bd679d72634dfb60a3d6b4bed29d15972232cc2068d0d22efce1c1518R110-R116

I tried to describe the existing Wasm configuration. However, only using CEL expressions from control plane.

Reply:
Sure this is similar to what I did, but I thought we could avoid `Matches` entirely given the routeRuleConditions will always be generated by Kuadrant and replaced it with `Predicates`

Reply:
Can we report in the status when people use them and _NOT_ apply the Policy? @guicassolato wdyt?

Reply:
That's for Auth btw, I think they should :fire: in RLP

Reply:
I think we should burn them in Auth as well actually. However, I'd do it as @adam-cattermole proposed it, i.e., by keeping the old user-defined `when` conditions (based on `selector, operator, value`) alongside with CEL as an addition only in a first iteration, generating the route conditions internally in CEL only (because why would we do it otherwise?), and then, in a separate PR, remove the old `when` conditions from both kinds of policy at once.

Reply:
Reporting in the status the use of old `when` conditions as invalid doesn't make sense to me. Why would we have them in the API if the user cannot use them?

Reply:
BTW, the amount of work to burn them in Auth in non-negligible. It's a lot of Authorino types to refine inside the Kuadrant Operator.

Reply:
Right... the amount of work is what annoys me 😮‍💨